### ISSUE TITLE
big model using admin diverge  

### ISSUE THREAD

**Issue by maydaygmail** [microsoft/admin-torch#1]
when use big model such as transformer_vaswani_wmt_en_de_big,  the encoder-decoder vary from 6-6, 18-18 or 24-24, loss will not diverge. All the settings are same as example/train_wmt_en-de_huge_batch.sh, except for arch is transformer_vaswani_wmt_en_de_big. The loss would decrease during the first 8 epochs, then the loss increase and diverge in the following epochs.

**Comment by LiyuanLucasLiu** [comment]
@namisan Thanks for the reminder, I didn't notice the issue before your assignment... 
@maydaygmail Thanks for bringing this up! If you have follow-up questions, please also at me to ensure I got the notification : )

**Typically, the learning rate needs to be scaled down, along with increasing the hidden state.**

In this case, since you change `embed-dim` from `512` to `1024` (`transformer_wmt_en_de` -> `transformer_vaswani_wmt_en_de_big`), we need to change the learning rate from `1.5e-3` to `7e-4` ($512 \cdot 1.5e-4 \approx 1024 \cdot 7e-4$).

For theoretical justifications/insights of this adaptation, there are some related analyses from the muP paper. It follows the same mechanism here. 

I also did some experiments to verify this and find the model converges well after changing the learning rate. Please find the commands, setting, and observations as below:


#### both O(logn) and O(1) with lr 7e-4 work well
the command to run O(logn) is:
```
bash train_big_wmt_en-de_huge_batch.sh \
  $PATH-to-WMT17
  24
  $OUTPUT_PATH_BIG_HUGE
  "--encoder-as-parameter --decoder-as-parameter --lr 0.0007"
```

the command to run O(1) is:
```
bash train_big_wmt_en-de_huge_batch.sh \
  $PATH-to-WMT17
  24
  $OUTPUT_PATH_BIG_HUGE
  "--encoder-as-parameter --decoder-as-parameter --lr 0.0007 --encoder-output-change-scale O(1) --decoder-output-change-scale O(1)"
```

#### O(1) with unadapted learning rate
I didn't finish the training, it seems that even for an unadapted learning rate (which is not a good practice), Admin seems to work well as well by setting the output change scale to O(1). 

The command to run experiments is:
```
# I didn't finish the training (not scaling lr is bad practice), but the loss reaches below 6 @ 10 epoch, which seems to work well. 
bash train_big_wmt_en-de_huge_batch.sh \
  $PATH-to-WMT17
  24
  $OUTPUT_PATH_BIG_HUGE
  "--encoder-as-parameter --decoder-as-parameter --lr 0.0015 --encoder-output-change-scale O(1) --decoder-output-change-scale O(1)"
```
